pkg/api/dashboard.go

```diff
@@ -101,9 +101,10 @@ func (hs *HTTPServer) GetDashboard(c *models.ReqContext) response.Response {
 	}
 
 	var (
-		hasPublicDashboard     = false
-		publicDashboardEnabled = false
-		err                    error
+		hasPublicDashboard                  = false
+		publicDashboardEnabled              = false
+		publicDashboardTimeSelectionEnabled = false
```

Comment:
I really don't understand why this is here, this is the normal dashboard load api route, it should know nothing about public dashboards, no need to load public dashboard settings.

This bit here should be removed. 
https://github.com/grafana/grafana/blob/main/pkg/api/dashboard.go#L111

maybe the check if it has a public dashboard or not could be kept, but should be all. We want to have isolation here, dashboard model should know nothing of settings of public dashboard.

Reply:
Talking about `publicDashboardTimeSelectionEnabled` you're totally right.

Talking about `hasPublicDashboard`, we decided to do that because we already need to make a db query to set publicDashboardEnabled (for Public tag in navbar) and we already have that value "for free". That value allows us not to make a unnecessary call to public dashboard data when the user clicks on the public dashboard modal. 
Moreover, because of the fact we're using backednSrvBaseQuery, if we make the api call and there's no public dashboard, a 404 will be returned and the backendSrvBaseQuery will automatically show an error tooltip. 
I know we can pass the showErrorAlert option as false but, in that case, we should handle the error and check if it's 404 or another kind of error in order to show or not the error tooltip. 
You're right we're coupling dashboard with public dashboard and that's not a good idea.
So, I'm gonna make the following change:
Make **always** the api call in public dashboard modal and handle the error, previously setting `showErrorAlert: false` in backendSrvBaseQuery